src/Functions/grouping.h

```diff
@@ -48,13 +53,15 @@ class FunctionGroupingBase : public IFunction
         auto result = ColumnUInt64::create();
         auto & result_data = result->getData();
         result_data.reserve(input_rows_count);
+
+        const auto * result_table = likely(force_compatibility) ? COMPATIBLE_MODE : INCOMPATIBLE_MODE;
```

Comment:
not an issue, but imo we should limit use of this intrinsic to hot loops when we also for some reason don't trust hw predictor